ph5/clients/ph5toms.py

```diff
@@ -508,44 +508,73 @@ def create_trace(self, station_to_cut, mp=False):
-
+                                       stc.endtime)
             if not das:
                 return
-
             das = [x for x in das]
             Das_tf = next(iter(das or []), None)
             if Das_tf is None:
                 return
             else:
                 das_t_start = (float(Das_tf['time/epoch_l']) +
                                float(Das_tf['time/micro_seconds_i']) / 1000000)
-
             if float(das_t_start) > float(stc.starttime):
                 start_time = das_t_start
 
             else:
                 start_time = stc.starttime
-
             nt = stc.notimecorrect
-
-            if stc.sample_rate > 0:
-                actual_sample_rate = float(
-                    stc.sample_rate) / float(stc.sample_rate_multiplier)
-            else:
-                actual_sample_rate = 0
-
-            if stc.sample_rate != 0:
+            for das_inst in das:
+                # Does Array.sr == DAS.sr? If so use sr
+                if das_inst['sample_rate_i'] == stc.sample_rate:
+                    if das_inst['sample_rate_i'] > 0:
+                        actual_sample_rate = float(stc.sample_rate) /\
+                                             float(stc.sample_rate_multiplier)
+                        empty_times = False
+                    else:
+                        actual_sample_rate = 0
+                        empty_times = False
+                else:
+                    continue
+            if empty_times is True:
+                for i, das_inst in enumerate(das):
+                    # Checks to see if all DAS tables have same SR
+                    sr_prev = das[i-1]['sample_rate_i']
+                    if das_inst['sample_rate_i'] != sr_prev:
+                        sr_mismatch = True
+                try:
+                    if sr_mismatch is True:
+                        # Else fail with error
+                        LOGGER.error('DAS and Array Table sample' +
+                                     ' rates do not match, DAS table' +
+                                     ' sample rates do not match.' +
+                                     ' Data must be updated.')
+                    else:
+                        # Uses DAS SR if consistent
+                        das_sr = das_inst['sample_rate_i']
+                        das_srm = das_inst['sample_rate_multiplier_i']
+                        LOGGER.warning('Using sample rate from' +
+                                       ' DAS Table. Sample rates' +
+                                       ' in DAS and Array tables' +
+                                       ' are not consistent.')
+                        if das_inst['sample_rate_i'] > 0:
+                            actual_sample_rate = float(das_sr) / float(das_srm)
+                        else:
+                            actual_sample_rate = 0
+                except(UnboundLocalError):
+                    continue
+            if actual_sample_rate != 0:
```

Comment:
I got error "local variable 'actual_sample_rate' referenced before assignment" when run this for the master file inside ph5/test_data/ph5/samplerate/error.

Reply:
Fixed in fix_424_help. All that need to be done was add return after log.error messages.